iroha/src/sumeragi.rs

```diff
@@ -2058,36 +2072,36 @@ mod tests {
                 queue,
                 sumeragi.clone(),
                 (events_sender.clone(), events_receiver),
-                None,
             );
             drop(task::spawn(async move {
                 torii.start().await.expect("Torii failed.");
             }));
             sumeragi.write().await.init(Hash([0_u8; 32]), 0);
             peers.push(sumeragi.clone());
             drop(task::spawn(async move {
-                while let Some(message) = sumeragi_message_receiver.next().await {
+                while let Some(message) = sumeragi_message_receiver.recv().await {
                     let _result = message.handle(&mut *sumeragi.write().await).await;
                 }
             }));
             let block_counter = block_counters[i].clone();
             drop(task::spawn(async move {
-                while let Some(_block) = block_receiver.next().await {
+                while let Some(_block) = block_receiver.recv().await {
                     *block_counter.write().await += 1;
                 }
             }));
         }
-        task::sleep(Duration::from_millis(2000)).await;
+        time::sleep(Duration::from_millis(2000)).await;
         // First peer is a leader in this particular case.
-        let leader = peers
-            .iter()
-            .find(|peer| {
-                task::block_on(async {
-                    let peer = peer.write().await;
-                    peer.network_topology.role(&peer.peer_id) == Role::Leader
-                })
-            })
-            .expect("Failed to find a leader.");
+        let mut leader = None;
+        for peer in peers {
+            let peer_read = peer.read().await;
+            if peer_read.network_topology.role(&peer_read.peer_id) == Role::Leader {
+                drop(peer_read);
+                leader = Some(peer);
+                break;
+            }
+        }
+        let leader = leader.expect("Failed to find leader.");
```

Comment:
Could you move this code into separate test function?

Reply:
If we want to do it, then it is better to do in a separate PR, I don't think we should focus on this in tokio migration.

Reply:
Before hand it was semantically just 1 function call, so logically it would be the same.

Reply:
Yes, but still this is out of scope of this PR to my mind. We can create a short PR after.

Reply:
https://jira.hyperledger.org/browse/IR-1108